### PR TITLE
currency synchronizer infrastructure

### PR DESCRIPTION
**Pull request by IvanLazitski**

### CODE REVIEW

File: Godel.CS.sln [hunk 1]
```diff
@@ -0,0 +1,73 @@
+﻿
```
Comment (Werocky1502) [inline Godel.CS.sln]: Remove obsolete solution Godel.CS

File: Godel.CS.Data.ViewModels/NBRate.cs [hunk 2]
```diff
@@ -0,0 +1,14 @@
+﻿using System;
+
+namespace Godel.CS.Data.ViewModels
+{
+    public class NbRate
```
Comment (Werocky1502) [inline Godel.CS.Data.ViewModels/NBRate.cs]: If you put it to ViewModels maybe would be better to rename it to RateViewModel ?

File: Godel.CS.Data.ViewModels/CurrencyInfo.cs [hunk 3]
```diff
@@ -0,0 +1,8 @@
+﻿namespace Godel.CS.Data.ViewModels
+{
+    public class CurrencyInfo
```
Comment (Werocky1502) [inline Godel.CS.Data.ViewModels/CurrencyInfo.cs]: What about removing of redundant "Info" part ?
What does it mean in this context ? It is a currency with name and code and nothing else, dosen't it ?

File: Godel.CS.BLL.Common/ICurrencyService.cs [hunk 4]
```diff
@@ -0,0 +1,12 @@
+﻿using System.Threading.Tasks;
+using Godel.CS.Data.ViewModels;
+
+namespace Godel.CS.BLL.Common
+{
+    public interface ICurrencyService
+    {
+        Task<NbRate> GetCurrencyUpdateAsync();
+        Task SendCurrencyInfoUpdateAsync(CurrencyInfo currencyInfo);
+        Task SendCurrencyRateUpdateAsync(CurrencyRate currencyRate);
```
Comment (Werocky1502) [inline Godel.CS.BLL.Common/ICurrencyService.cs]: If I'm not mistaken our currency manager should receive CurrenciesList and CurrenciesRates. What about renaming your CurrencyInfoUpdate and CurrencyRateUpdate signatures.
And why did you create GetCurrencyUpdate and do not create GetRateUpdate ?

Maybe would be better to remove those methods at all they are not a part of current task.

File: Godel.CS.Data.ViewModels/CurrencyInfo.cs [hunk 5]
```diff
@@ -0,0 +1,8 @@
...
+    {
+        public string Code { get; set; }
+        public string Name { get; set; }
```
Comment (VladDalidovich) [inline Godel.CS.Data.ViewModels/CurrencyInfo.cs]: Whitespace between prop

File: Godel.CS.Data.ViewModels/NBRate.cs [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+    {
+        public int Cur_ID { get; set; }
+
+        public DateTime Date { get; set; }
```
Comment (VladDalidovich) [inline Godel.CS.Data.ViewModels/NBRate.cs]: wrong naming

File: Godel.CS.BLL.Common/ICurrencyService.cs (hunk 4 above)
Comment (IvanLazitski) [reply Godel.CS.BLL.Common/ICurrencyService.cs]: Agree with renaming CurrencyInfoUpdate to CurrencyListUpdate. We will send only one request to national bank, and in it response with all information that we need.

File: Godel.CS.BLL.Common/ICurrencyService.cs [hunk 7]
```diff
@@ -0,0 +1,12 @@
+﻿using System.Threading.Tasks;
+using Godel.CS.Data.ViewModels;
+
+namespace Godel.CS.BLL.Common
+{
+    public interface ICurrencyService
+    {
+        Task<NbRate> GetCurrencyUpdateAsync();
+        Task SendCurrencyListUpdateAsync(Currency currency);
```
Comment (Werocky1502) [inline Godel.CS.BLL.Common/ICurrencyService.cs]: Would you send one Currency, or many of them?
The same to the hole interface.

**Pull request merged**